### PR TITLE
[deps]: update jwt-go dependency

### PR DESCRIPTION
**Pull request by emirisman**
## Description

jwt-go has been renamed to jwt and has a new home. See
https://github.com/dgrijalva/jwt-go/issues/462

## Checklist:
- [x] Documentation updated

### CODE REVIEW

File: CREDITS [hunk 1]
```diff
@@ -3770,10 +3770,11 @@ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 
 ================================================================
 
-github.com/dgrijalva/jwt-go
-https://github.com/dgrijalva/jwt-go
+github.com/golang-jwt/jwt
+https://github.com/golang-jwt/jwt
 ----------------------------------------------------------------
 Copyright (c) 2012 Dave Grijalva
+Copyright (c) 2021 jwt-go maintainers
```
Comment (harshavardhana) [inline CREDITS]: is this manually edited? @emirisman

File: cmd/jwt.go [hunk 2]
```diff
@@ -22,8 +22,8 @@ import (
 	"net/http"
 	"time"
 
-	jwtgo "github.com/dgrijalva/jwt-go"
-	jwtreq "github.com/dgrijalva/jwt-go/request"
+	jwtgo "github.com/golang-jwt/jwt"
+	jwtreq "github.com/golang-jwt/jwt/request"
```
Comment (harshavardhana) [inline cmd/jwt.go:26]: Before we do this, I think the `golang-jwt/jwt` should fix the incompatible issue with go.mod

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/jwt.go (hunk 2 above)
Comment (emirisman) [reply cmd/jwt.go:26]: They have added go modules support but had to revert it. See https://github.com/golang-jwt/jwt/pull/26 and https://github.com/golang-jwt/jwt/issues/17 for details. It doesn't have any external dependencies itself therefore there are no downsides except the visually unpleasing `+incompatible` suffix

File: CREDITS (hunk 1 above)
Comment (emirisman) [reply CREDITS]: It is the current LICENSE file https://github.com/golang-jwt/jwt/blob/main/LICENSE

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: CREDITS (hunk 1 above)
Comment (emirisman) [reply CREDITS]: oh my bad, copied the wrong string

**Review by harshavardhana — APPROVED** [review]

File: go.sum [hunk 3]
```diff
@@ -533,6 +533,8 @@ github.com/gogo/protobuf v1.2.2-0.20190723190241-65acae22fc9d/go.mod h1:SlYgWuQ5
 github.com/gogo/protobuf v1.3.1/go.mod h1:SlYgWuQ5SjCEi6WLHjHCa1yvBfUnHcTbrrZtXPKa29o=
 github.com/gogo/protobuf v1.3.2 h1:Ov1cvc58UF3b5XjBnZv7+opcTcQFZebYjWzi34vdm4Q=
 github.com/gogo/protobuf v1.3.2/go.mod h1:P1XiOD3dCwIKUDQYPy72D8LYyHL2YPYrpS2s69NZV8Q=
+github.com/golang-jwt/jwt v3.2.1+incompatible h1:73Z+4BJcrTC+KczS6WvTPvRGOp1WmfEP4Q1lOd9Z/+c=
+github.com/golang-jwt/jwt v3.2.1+incompatible/go.mod h1:8pz2t5EyA70fFQQSrl6XZXzqecmYZeUEB8OUGHkxJ+I=
 github.com/golang-sql/civil v0.0.0-20190719163853-cb61b32ac6fe/go.mod h1:8vg3r2VgvsThLBIFL93Qb5yWzgyZWhEmBwUJWevAkK0=
 github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b h1:VKtxabqXZkF25pY9ekfRL6a582T4P37/31XEstQ5p58=
 github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b/go.mod h1:SBH7ygxi8pfUlaOkMMuAQtPIUF8ecWP5IEl/CR7VP2Q=
```
Comment (harshavardhana) [inline go.sum:540]: can you run `go mod tidy` ?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: go.sum (hunk 3 above)
Comment (emirisman) [reply go.sum:540]: It is included, `go mod tidy` doesn't make further changes

Comment (harshavardhana) [reply go.sum:540]: Okay..

**Review by harshavardhana — APPROVED** [review]

**Comment by emirisman** [comment]
Rebased on master. Looks like the CREDITS is autogenerated so I'm not touching it

**Review by harshavardhana — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by klauspost — APPROVED** [review]
LGTM

**Pull request merged**